src/main/java/com/github/onsdigital/babbage/search/external/requests/base/AbstractSearchRequest.java

```diff
@@ -53,10 +71,20 @@ protected Request post() throws Exception {
      */
     protected abstract ContentResponse getContentResponse() throws Exception;
 
+    public String getContentResponseAsString() throws Exception {
+        return this.getContentResponse().getContentAsString();
+    }
+
     @Override
     public T call() throws Exception {
-        String response = this.getContentResponse().getContentAsString();
-        return MAPPER.readValue(response, this.returnClass);
+        String response = this.getContentResponseAsString();
+
+        // Either typeReference or returnClass are guaranteed to not be null
+        if (this.typeReference != null) {
+            return MAPPER.readValue(response, this.typeReference);
+        } else {
+            return MAPPER.readValue(response, this.returnClass);
```

Comment:
Super picky comment but `else` in this case is not required - if the `if` case is true the method returns, you can replace the else block with `return MAPPER.readValue(response, this.returnClass);`